addons/l10n_in_edi/models/res_company.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    l10n_in_edi_username = fields.Char("E-invoice (IN) Username", groups="base.group_system")
+    l10n_in_edi_password = fields.Char("E-invoice (IN) Password", groups="base.group_system")
+    l10n_in_edi_token = fields.Char("E-invoice (IN) Token", groups="account.group_system")
+    l10n_in_edi_token_validity = fields.Datetime("E-invoice (IN) Valid Until", groups="account.group_system")
+    l10n_in_edi_production_env = fields.Boolean(
+        string="E-invoice (IN) Is production OSE environment",
+        help="Enable the use of production credentials",
+        groups="base.group_system",
+    )
+
+    def _l10n_in_edi_token_is_valid(self):
+        self.ensure_one()
+        if self.l10n_in_edi_token and self.l10n_in_edi_token_validity > fields.Datetime.now():
+            return True
+        return False
+
+    def _neutralize(self):
+        super()._neutralize()
+        self.flush()
+        self.invalidate_cache()
+        self.env.cr.execute("UPDATE res_company SET l10n_in_edi_production_env = true")
```

Comment:
when is this called? is `true` the right value?

Reply:
@mart-e ohh, it's a mistake because first, we have a field like l10n_in_edi_test_env
From v15 we removed this _neutralize but we set this to `false` in master
Thank you